### PR TITLE
[MNT] change cycle (0.29.0) for renaming `cINNForecaster` to `CINNForecaster`

### PR DESCRIPTION
**Pull request by geetu040**
#### Reference Issues/PRs
This PR completes step 2 of change cycle for renaming cINNForecaster to CINNForecaster completing the steps for release v0.29.0 in https://github.com/sktime/sktime/issues/6120

#### What does this implement/fix? Explain your changes.
- renames the classes
- updates deprecation warning

### CODE REVIEW

**Comment by fkiraly** [comment]
great! scheduled for 0.29.0 then

File: sktime/forecasting/conditional_invertible_neural_network.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-"""Conditional Invertible Neural Network (cINN) for forecasting."""
```
Comment (fkiraly) [inline sktime/forecasting/conditional_invertible_neural_network.py:1]: I think the docstring name can stay, thats how it is in the paper

Comment (geetu040) [reply sktime/forecasting/conditional_invertible_neural_network.py]: and what about the class docstring? I also changed that to `CINN`

Comment (fkiraly) [reply sktime/forecasting/conditional_invertible_neural_network.py]: I'd leave that `cINN`, I think that's consistent with the publications. The issue is only with the class name, which is not PEP compliant.

**Pull request merged**